source/transactions-convenient-api/tests/callback-commits.yml

```diff
@@ -97,6 +97,8 @@ tests:
     # Session state will be NO_TXN when callback returns to withTransaction
     description: withTransaction still succeeds if callback commits and runs extra op
     useMultipleMongoses: true
+    clientOptions:
+      retryWrites: false
```

Comment:
Likewise, the APM assertion for the final insert after the commit avoids asserting `txnNumber`.

Reply:
Please add a comment explaining why this is necessary and remove the "This test is agnostic about retryWrites" comment further down in this file.